packages/react-dom/src/__tests__/ReactDOMStrictEffects-test.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @emails react-core
+ */
+
+'use strict';
+
+let React;
+let ReactDOMClient;
+let ReactDOMServer;
+let Scheduler;
+let act;
+let assertLog;
+
+describe('StrictEffectsMode', () => {
+  beforeEach(() => {
+    jest.resetModules();
+    act = require('internal-test-utils').act;
+    const InternalTestUtils = require('internal-test-utils');
+    assertLog = InternalTestUtils.assertLog;
+
+    React = require('react');
+    Scheduler = require('scheduler');
+    ReactDOMClient = require('react-dom/client');
+    ReactDOMServer = require('react-dom/server');
+  });
+
+  // @gate !useModernStrictMode
+  it('should double invoke effects during hydration', async () => {
+    function App({text}) {
+      React.useEffect(() => {
+        Scheduler.log('useEffect mount');
+        return () => Scheduler.log('useEffect unmount');
+      });
+
+      return null;
+    }
+    const container = document.createElement('div');
+    const element = (
+      <React.StrictMode>
+        <App />
+      </React.StrictMode>
+    );
+    container.innerHTML = ReactDOMServer.renderToString(element);
+
+    await act(() => {
+      ReactDOMClient.hydrateRoot(container, element);
```

Comment:
Uses `mountChildFibers` which has side-effect tracking disabled. Side-effect tracking sets the `PlacementDEV` flag. Since `PlacementDEV` is never set here, we never double invoke: https://github.com/facebook/react/blob/94eed63c49d989861ae7cd62e111de6d717f0a10/packages/react-reconciler/src/ReactFiberWorkLoop.js#L3691-L3701

But then it's not clear yet why we need this flag at all. Shouldn't we double invoke regardless of whether something was placed?